sqlitecluster/SQLite.cpp

```diff
@@ -256,79 +310,146 @@ bool SQLite::write(const string& query) {
     return true;
 }
 
-// --------------------------------------------------------------------------
 bool SQLite::prepare() {
     SASSERT(_insideTransaction);
+
+    // We lock this here, so that we can guarantee the order in which commits show up in the database.
+    commitLock.lock();
+
+    // Now that we've locked anybody else from committing, look up the state of the database.
+    string committedQuery, committedHash;
+    uint64_t commitCount = _commitCount.load();
```

Comment:
Just curious, should we be using `auto` in scenarios like this?  Generalizes our code a bit to not be tightly coupled to the return value type.  Not insisting on it, just trying to figure out a best practice.